datalad/distribution/get.py

```diff
@@ -871,7 +891,7 @@ def __call__(
                 # as is
                 dataset=dataset,
                 # always come from the top to get sensible generator behavior
```

Comment:
so this change changes this behavior:
- I wonder why top-down was considered to be more "sensible" (may be smth in commit which added)
- comment should be adjusted according to the change if this change is needed (seems to be orthogonal to the actual fix described)

Reply:
Thanks for picking this up, I missed the comment and will update